### PR TITLE
Overloading the modulo operator for doubles and longs

### PR DESCRIPTION
**Pull request by m-schindler**
This change would avoid confusions between the % operator and the fmod function in the future. See the discussion at https://community.freefem.org/t/laplace-eigenproblem-with-robin-boundary-condition/895/

### CODE REVIEW

File: src/bamglib/write_xdmf.cpp [hunk 1]
```diff
@@ -67,7 +67,7 @@ void WriteXdmf::WriteXdmfMeshFile2D( ) {
   xdmf_file << "</Xdmf>\n";
   xdmf_file.close( );
 
-  cout << "save xdmf file mesh : " << xdmf_filename << endl;
+  //cout << "save xdmf file mesh : " << xdmf_filename << endl;
```
Comment (prj-) [inline src/bamglib/write_xdmf.cpp:70]: No dead code please, you can just remove this line.

**Comment by prj-** [comment]
I've changed the base branch to `develop` since we don't allow direct contribution to `master`, hope you don't mind.

**Comment by prj-** [comment]
You are not bothering me, thank you for your contribution. I'll review that more thoroughly tomorrow probably.

File: src/bamglib/write_xdmf.cpp (hunk 1 above)
Comment (prj-) [reply src/bamglib/write_xdmf.cpp:70]: Could you please fix that?
```suggestion
```

File: src/fflib/AFunction.cpp [hunk 2]
```diff
@@ -1241,7 +1241,10 @@ void Init_map_type()
        );
 
      TheOperators->Add("%",
-       new OneBinaryOperator<Op2_mod<long,long,long> >
+       new OneBinaryOperator<Op2_mod<long,long,long> >,
+       new OneOperator2<double,double,double>(fmod),
+       new OneOperator2<double,long,double>(fmod),
+       new OneOperator2<double,double,long>(fmod)
```
Comment (prj-) [inline src/fflib/AFunction.cpp:1247]:
```suggestion
       new OneOperator2<double,double,double>(fmod)
```
I don't think it's necessary to add these casts.

**Review by prj- — COMMENTED** [review]
I've tried your branch, it works nicely, could you please apply the suggested changes?

**Comment by frederichecht** [comment]
oui, pour moi,

**Comment by prj-** [comment]
Thank you for your contribution.

**Comment by m-schindler** [comment]
Thank you for accepting my contribution.

**Comment by prj-** [comment]
OK, so in fact, your branch is not compiling... I will fix that shortly.

**Comment by m-schindler** [comment]
On Thu, Apr 29, 2021 at 06:47:33AM -0700, prj- wrote:
> OK, so in fact, your branch is not compiling... I will fix that shortly.

I forgot the declaration of "verbosity" in src/bamglib/write_xdmf.hpp
I corrected this.

**Pull request merged**